Update Sponsored Fees docs with latest changes (#223)
ripple/opensource.ripple.com: docs/xls-68-sponsored-fees-and-reserves/references/transactions/sponsorshipset.md

| `RemainingOwnerCount` | Number    | UInt32            | No        | The remaining amount of reserves that the sponsor has provided for the sponsee to use. This value replaces what is currently in the `Sponsorship.RemainingOwnerCount` field, if it exists. |
| `Sponsee`             | String    | AccountID         | No        | The sponsee associated with this relationship. If this field is included, the `Account` is assumed to be the sponsor. |

A sponsorship entry must keep at least some fee budget (`FeeAmount`) or reserve budget (`RemainingOwnerCount`). If you try to create or update the entry with neither, the transaction fails, because an empty entry still consumes an owner reserve while giving the sponsee nothing to draw on.

{% admonition type="info" name="Note" %}
`FeeAmount` is a replacement value rather than an increment, meaning the new value overwrites the old one. Raising `FeeAmount` deducts the additional XRP from the sponsor, while lowering it refunds the difference back to the sponsor.

> {% /admonition %}

Reviewer: I think we can just include this info in the description column.

| Error Code        | Description |
| :---------------- | :---------- |
| `tecNO_DST`       | The sponsor or sponsee account does not exist on the ledger. |
| `tecNO_ENTRY`     | The `tfDeleteObject` flag is enabled but the `Sponsorship` ledger entry does not exist. |
| `tecNO_PERMISSION` | The transaction is not permitted. This can occur when:<ul><li>Creating or updating the entry would leave it with no budget (neither a positive `FeeAmount` nor a positive `RemainingOwnerCount`). Empty `Sponsorship` ledger entries are not allowed.</li><li>The sponsor or sponsee is a pseudo-account, which cannot participate in sponsorship.</li></ul> |
| `tecUNFUNDED`     | The sponsor does not have sufficient XRP to fund the `FeeAmount` or to cover the reserve for the `Sponsorship` ledger entry. |
| `temBAD_AMOUNT`   | An amount field is invalid. This can occur when:<ul><li>`FeeAmount` is negative or not denominated in XRP.</li><li>`MaxFee` is negative or not denominated in XRP.</li></ul> |

> | `temINVALID_FLAG` | The transaction has invalid flags. This can occur when:<ul><li>Conflicting flags are enabled, such as both `tfSponsorshipSetRequireSignForFee` and `tfSponsorshipClearRequireSignForFee`.</li><li>The `tfDeleteObject` flag is enabled with flags that modify the entry's settings.</li></ul> |

Reviewer: ```suggestion
| `temINVALID_FLAG` | The transaction has invalid flags. This can occur when:<ul><li>Conflicting flags are enabled, such as both `tfSponsorshipSetRequireSignForFee` and `tfSponsorshipClearRequireSignForFee`.</li><li>The `tfDeleteObject` flag is enabled along with additional flags. `tfDeleteObject` must be the only flag.</li></ul> |
```
